tests/integration_test/services_test.py

```
@@ -2298,6 +2304,265 @@
+
+@patch("kairon.api.app.routers.bot.data.UploadHandler")
+def test_upload_file_content_no_enqueue_when_validate_false(mock_upload_handler):
+    """Test that file upload does not enqueue event when validation fails."""
+
+    # Arrange
+    mock_event_instance = MagicMock()
+    mock_event_instance.validate.return_value = False
+    mock_upload_handler.return_value = mock_event_instance
+
+    files = {
+        "file_content": ("Salesstore.csv", open("tests/testing_data/file_content_upload/Salesstore.csv", "rb"))
```

Ruff (0.12.2)

Use a context manager for opening files

(SIM115)